container/service.go

```diff
@@ -1,137 +1,106 @@
 package container
 
 import (
-	"bytes"
 	"context"
+	"io"
 
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
 	"github.com/docker/docker/api/types/swarm"
-	godocker "github.com/fsouza/go-dockerclient"
+	docker "github.com/docker/docker/client"
 )
 
 // ListServices returns existing docker services matching a specific label name
-func ListServices(label string) (dockerServices []swarm.Service, err error) {
+func ListServices(label string) (services []swarm.Service, err error) {
 	client, err := Client()
 	if err != nil {
 		return
 	}
-	dockerServices, err = client.ListServices(godocker.ListServicesOptions{
-		Context: context.Background(),
-		Filters: map[string][]string{
-			"label": []string{label},
-		},
+	services, err = client.ServiceList(context.Background(), types.ServiceListOptions{
+		Filters: filters.NewArgs(filters.KeyValuePair{
+			Key:   "label",
+			Value: label,
+		}),
 	})
-	if err != nil {
-		return
-	}
 	return
 }
 
-//  FindService returns the Docker Service
-func FindService(namespace []string) (dockerService *swarm.Service, err error) {
+// FindService returns the Docker Service. Return error if not found.
+func FindService(namespace []string) (service swarm.Service, err error) {
 	client, err := Client()
 	if err != nil {
 		return
 	}
-	dockerServices, err := client.ListServices(godocker.ListServicesOptions{
-		Filters: map[string][]string{
-			"name": []string{Namespace(namespace)},
-		},
-		Context: context.Background(),
-	})
-	if err != nil {
-		return
-	}
-	for _, service := range dockerServices {
-		if service.Spec.Name == Namespace(namespace) {
-			dockerService = &service
-			return
-		}
-	}
+	service, _, err = client.ServiceInspectWithRaw(context.Background(), Namespace(namespace), types.ServiceInspectOptions{})
 	return
 }
 
 // StartService starts a docker service
-func StartService(options godocker.CreateServiceOptions) (dockerService *swarm.Service, err error) {
+func StartService(options ServiceOptions) (serviceID string, err error) {
 	client, err := Client()
 	if err != nil {
 		return
 	}
-	return client.CreateService(options)
+	options.merge()
```

Comment:
I really don't like this for many reasons
- first you don't know what you are merging here `option.merge()` but you give no parameters so it looks like you merge nothing so this function looks useless.
- also this kind of function should return something, merge means there is a modification in something so you need to get the update (like the function `append` for example)
- finally it's just not clear what this is doing and there is no reason to call this method if you don't know that you have to call it...

My proposition:

Rename this function with a `option.toDockerService()` that returns the object needed for docker like that the function name is clear and we can give the result directly in parameter of the `ServiceCreate` function and it's easier to test etc...